index.js

```diff
@@ -54,8 +58,30 @@ module.exports = function(glob, options) {
   return ing.negated ? '!' + glob : glob;
 };
 
+function escape(path) {
+  return path
+    .replace(/\[(.*?)\]/g, "\\[$1\\]")
+    .replace(/\((.*?)\)/g, "\\($1\\)")
+    .replace(/\{(.*?)\}/g, "\\{$1\\}")
+    .replace(/\*/g, "\\*")
+    .replace(/\!/g, "\\!")
+    .replace(/\?/g, "\\?");
+}
+
+function unescape(path) {
+  return path
+    .replace(/\\\[(.*?)\\\]/g, "[$1]")
+    .replace(/\\\((.*?)\\\)/g, "($1)")
+    .replace(/\\\{(.*?)\\\}/g, "{$1}")
+    .replace(/\\\*/g, "*")
+    .replace(/\\\!/g, "!")
+    .replace(/\\\?/g, "?");
+}
```

Comment:
hey, @phated! hope you're doing well. 

This makes sense. I think we should be able to simplify the replacements to the following:

```js
function escape(path) {
  return path.replace(/([({[\]})*?!])/g, '\\$1');
}

function unescape(path) {
  return path.replace(/\\([({[\]})*?!])/g, '$1');
}
```

Reply:
Things are well! Hope the same for you.

I think your change makes sense for `unescape` (since we want to remove all escaping before doing normalization) but I think we only want to escape if we have a matching pair of grouping operators, such as `( )` or `[ ]`. For that, should I keep `escape` implementation mostly the same as it is?

Reply:
This example shows the challenge of trying to match wrapping/bracketed characters:

```js
// Since we're trying to match both sides, using `?` (in `.*?`) causes the regex parser to lazily 
// match up to the first `}`,  which then causes the second pair to not be matched. 
// fwiw, we'd have the same issue if we removed `?` from `.*?`, since it would match the first 
// and last characters, but not the middle
console.log(escape('aaa/bbb{{xxx}}ccc/ddd'));
// aaa/bbb\{{xxx\}}ccc/ddd

// The only way we can be sure to match all of them is to either parse the path and use a 
// stack, or just escape all of them. Also it shouldn't cause any matching problems if we
// escape non-wrapping bracket chars as well. 
console.log(escape2('aaa/bbb{{xxx}}ccc/ddd'));
// aaa/bbb\{\{xxx\}\}ccc/ddd
```

Reply:
Even if it's unlikely that people will use nested brackets in paths, odds are it'll still happen and ppl will create issues about it. But escaping extra characters will just cause those characters to be treated as literals by the matcher, so it shouldn't hurt anything.

Reply:
Ah, good point! I agree that "over-escaping" is better than having a weird edge case. I'll make those changes.